pkg/stub/handler.go

```diff
@@ -902,68 +1003,114 @@ func (h *Handler) Handle(ctx context.Context, event sdk.Event) error {
 		// 2) then after we remove finalizer, comes in with delete timestamp
 		// and event delete flag true
 		if event.Deleted {
-			// possibly tell poller to stop
 			logrus.Info("A previous delete attempt has been successfully completed")
 			return nil
 		}
 		if srcfg.DeletionTimestamp != nil {
+			// before we kick off the delete cycle though, we make sure a prior creation
+			// cycle is not still in progress, because we don't want the create adding back
+			// in things we just deleted ... if a create is still in progress, return an
+			// error so we retry
+			if srcfg.ConditionTrue(v1alpha1.ImageChangesInProgress) {
+				return fmt.Errorf("Deletes will not be processed until outstanding creates are completed")
```

Comment:
why do we need to retry?  won't we get another update event when the imagechangesinprogress status gets updated to false?

Reply:
yeah I was hesitant to depend on the update from elsewhere .. but I believe you are correct

I'll try the change out, verify things still look OK